xoinvader/background.py

```diff
@@ -0,0 +1,261 @@
+    :type filename: string
+
+    :param trim_width: width to truncate long lines to
+    :type trim_width: integer"""
+
+    with open(filename, "r") as f:
+        chunks = []
+        current_chunk = None
+        names = []
+        for line in f:
+            line = line.rstrip('\n')
+            if not line:
+                continue
+            elif line.startswith(Chunk.__MAGIC__):
+                name_split = line.split()
+                if len(name_split) < 2:
+                    raise ValueError("Chunks with empty names are not allowed")
+                name = name_split[1].strip()
+                if name in names:
+                    raise ValueError("Name {} already defined in background "
+                                     "file {}".format(name, filename))
+                current_chunk = Chunk(name)
+                chunks.append(current_chunk)
+                names.append(name)
+                continue
+            elif not current_chunk:
+                continue
+            # ok, now we got both chunk and something to store in it
+            if trim_width:
+                line_add = line[:trim_width]
+            else:
+                line_add = line
+            current_chunk.add_line(line_add)
+        if not chunks:
+            raise ValueError("File {} does not contain "
+                             "any chunks".format(filename))
+        return chunks
+
+
+class Background(Renderable):
+    """Class for storing and managing level background.
+    Manages background rendering and changing in time.
+
+    :param loop: flag if current chunk should loop after it had ended
+    :type loop: boolean
+
+    :param loop_all: flag if whole background should repeat from beginning
+    :type loop: boolean
+
+    :param speed: speed of background advance. Can be negative to represent
+    moving backwards
+    :type speed: integer
+
+    :param chunks: list of background chunks
+    :type chunks: list
+
+    :param background: list of background lines. You may change it at any time.
+    This list directly converts into Surface which then goes to the renderer
+    :type list:
+
+    """
+    render_priority = -1000
+
+    def __init__(self, filename=None):
+        self.loop = False
+        self.loop_all = False
+        self.speed = 0
+        self.chunks = []
+        self.background = []
+
+        self._w = Settings.layout.field.edge.x
+        self._h = Settings.layout.field.edge.y
+        self._background_surface = None
+        self._current_chunk = None
+        self._current_chunk_num = 0  # position of chunk in chunk list
+        self._chunk_line = 0  # position in current chunk
+        self._ticks_since_last_update = 0
+
+        if filename:
+            self.load_file(filename)
+
+    def clear(self):
+        """Clear all background.
+        Doesn't prevent updating, if speed is not 0."""
+        self.background = [' ' * self._w] * self._h
+
+    def load_file(self, filename):
```

Comment:
I know that this is obvious, but add docstring too please) This is for documetation generation too.